content/tutorial-svd.md

```diff
@@ -202,9 +202,9 @@ results in a `ValueError`. This happens because having a one-dimensional array f
 
 ```{code-cell} ipython3
 import numpy as np
-Sigma = np.zeros((768, 1024))
-for i in range(768):
-    Sigma[i, i] = s[i]
+Sigma = np.zeros((U.shape[0], Vt.shape[0]))
+diag_indices = (np.arange(U.shape[0]),) * 2
+Sigma[diag_indices] = s
```

Comment:
This recommended change is a bit hairy: I basically thought it was an opportunity to introduced the advanced indexing concept of using arrays of indices. However I totally understand if this muddies the waters here too much. If the advanced indexing is too complicated, it may worth swapping the for-loop out with `np.fill_diagonal`, since it is used later in the tutorial anyways.

It's also totally fine as is, just something that jumped out to me as an opportunity to highlight more numpy features!

Reply:
I think this is fine as long as you mention it first - maybe write a small warning before the operations explaining what you are about to do? This can be very surprising if the user is not very familiar with NumPy - which is the assumption here.

Reply:
I went with @rkern 's suggestion on this one (see comment below, which I'll leave unresolved).